### PR TITLE
created AuthenticationService, HashUtil, User, UserDao, UserService c…

### PR DESCRIPTION
**Pull request by Alekearl**
…lasses

### CODE REVIEW

File: src/main/java/com/dev/cinema/exception/AuthenticationException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.dev.cinema.exception;
+
+public class AuthenticationException extends Throwable {
```
Comment (oleksandrdziubiuk) [inline src/main/java/com/dev/cinema/exception/AuthenticationException.java]: можна наслідуватись від Exception

File: src/main/java/com/dev/cinema/model/User.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public void setSalt(byte[] salt) {
+        this.salt = salt;
+    }
+
+    @Override
+    public String toString() {
+        return "User{"
+                + "id=" + id
+                + ", email='" + loginByEmail + '\''
+                + ", password='" + password + '\''
```
Comment (oleksandrdziubiuk) [inline src/main/java/com/dev/cinema/model/User.java]: password не треба виводити на екран

File: src/main/java/com/dev/cinema/service/AuthenticationService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.dev.cinema.service;
+
+import com.dev.cinema.exception.AuthenticationException;
+import com.dev.cinema.model.User;
+
+public interface AuthenticationService {
+    User login(String email, String password) throws AuthenticationException;
+
+    User register(String email, String password);
+
```
Comment (oleksandrdziubiuk) [inline src/main/java/com/dev/cinema/service/AuthenticationService.java]: пуста строка

**Review by oleksandrdziubiuk — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/daoimpl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
     public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             Query<MovieSession> getAvailableMovieSessionsQuery = session
-                    .createQuery("FROM MovieSession WHERE movie.id = :movieId "
+                    .createQuery("FROM MovieSession WHERE id = :movieId "
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/daoimpl/MovieSessionDaoImpl.java]: why have you changed this?

File: src/main/java/com/dev/cinema/dao/daoimpl/UserDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> userEmailQuery =
+                    session.createQuery("FROM User WHERE loginByEmail = :loginByEmail", User.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/daoimpl/UserDaoImpl.java]:
```suggestion
                    session.createQuery("FROM User WHERE email= :email", User.class);
```

File: src/main/java/com/dev/cinema/exception/AuthenticationException.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/exception/AuthenticationException.java]: I agree with Oleksandr

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 6]
```diff
@@ -9,15 +9,13 @@
 import javax.persistence.Table;
 
 @Entity
-@Table(name = "movie session")
+@Table(name = "movie_session")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/MovieSession.java:12]:
```suggestion
@Table(name = "movie_sessions")
```
if you choose to name table for User `users` - let's have plural for **all** other tables as well

File: src/main/java/com/dev/cinema/model/User.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(unique = true)
+    private String loginByEmail;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/User.java]: check the UML and name this field accordingly please

File: src/main/java/com/dev/cinema/model/User.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/model/User.java]: I agree

File: src/main/java/com/dev/cinema/service/AuthenticationService.java (hunk 3 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/service/AuthenticationService.java]: agree

File: src/main/java/com/dev/cinema/model/User.java (hunk 2 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/User.java]: Ok. Changed.

File: src/main/java/com/dev/cinema/service/AuthenticationService.java (hunk 3 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/service/AuthenticationService.java]: Removed.

File: src/main/java/com/dev/cinema/exception/AuthenticationException.java (hunk 1 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/exception/AuthenticationException.java]: Refactored.

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/daoimpl/MovieSessionDaoImpl.java (hunk 4 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/dao/daoimpl/MovieSessionDaoImpl.java]: I've had problems with sql query in DB, so i searched and refactored code. While were trying to solve one problem, deleted many-to-one movie field here. Haven't noticed it, thanx for paying attention to this.

File: src/main/java/com/dev/cinema/dao/daoimpl/UserDaoImpl.java (hunk 5 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/dao/daoimpl/UserDaoImpl.java]: Ok.

File: src/main/java/com/dev/cinema/model/MovieSession.java (hunk 6 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/MovieSession.java:12]: I'll change users table to user. Had problems with sql db connection and queries in some classes, so tried figure out what's wrong and where.

File: src/main/java/com/dev/cinema/model/User.java (hunk 7 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/User.java]: Ok.

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**